### DISCUSSION TITLE
Given EdgeDB stores data as a graph, will EdgeDB make it possible to add graph algorithms?

### DISCUSSION THREAD

**Discussion by benwoodward** [geldata/gel#3411] [Q&A] [answered]
For example: https://neo4j.com/developer/graph-data-science/graph-algorithms/

**Comment by elprans** [comment, marked as answer]
Hi @benwoodward!  Thanks for your question!

EdgeDB is not a graph database, at least not in the classical sense like neo4j.  

EdgeDB does not _store_ data as a graph, it _models_ data as a graph using what we call a *graph-relational* data model.  Graph-relational is a reformulation of the classical relational model that makes several key modifications and additions to eliminate the object-relational impedance mismatch _completely_.  Data is still stored mostly relationally, but is modeled and queried as a *strongly-typed* semantic graph.  

This means that you can write queries like `SELECT Object FILTER Object.link1.link2.link3.link4 = <value>` or `SELECT Object { link1: { link2: { props } } }`, and, for certain cases, things like k-nearest neighbor and transitive closures are also possible.  For more advanced graph operations, especially where there is no set schema, EdgeDB would not be the ideal tool.

**Reply by benwoodward** [reply]
Thanks
